src/components/Save.js

```diff
@@ -11,28 +11,29 @@ import * as selectors from '../redux/selectors';
 //TODO: Link argument of saveGoogleDrive() to Input Value
 export class Save extends React.Component {
   render() {
-    return (
-      <span>
-        {this.props.hasConnectedDrive && !this.props.isSavingDrive &&
-         <span>
-           <Input kind="program" defaultValue="Program Name"></Input>
-           <Button kind="toolbar"
-                   onClick={() => this.props.saveGoogleDrive("File")}>
-             Save
-           </Button>
-         </span>
-        }
-        {this.props.hasConnectedDrive && this.props.isSavingDrive &&
-         <span>
-           <Input kind="program" defaultValue="Program Name"></Input>
-           <Button kind="toolbar">
-             <Spinner style={styles.spinners.toolbar}/>
-             Saving...
-           </Button>
-         </span>
-        }
-      </span>
-    );
+    if (!this.props.hasConnectedDrive) {
+      return null;
+    } else if (this.props.isSavingDrive) {
```

Comment:
You don't need to put these into `else if`s. The else is implied since there is a return in the above if statement. The common pattern goes something like this:
```javascript
function myFunction(someParams) {
  if (checkEarlyExitCondition()) {
    return;
  }
  ... actual body of the function ...
}
```

Here's some more discussion of this style: http://programmers.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement